pkg/query-service/app/logparsingpipeline/preview.go

```diff
@@ -0,0 +1,254 @@
+package logparsingpipeline
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/logstransformprocessor"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/processor"
+	"go.signoz.io/signoz/pkg/query-service/collectorsimulator"
+	"go.signoz.io/signoz/pkg/query-service/model"
+	"gopkg.in/yaml.v3"
+)
+
+func SimulatePipelinesProcessing(
+	ctx context.Context,
+	pipelines []Pipeline,
+	logs []model.SignozLog,
+) (
+	[]model.SignozLog, *model.ApiError,
+) {
+
+	if len(pipelines) < 1 {
+		return logs, nil
+	}
+
+	// Collector simulation does not guarantee that logs will come
+	// out in the same order as in the input.
+	//
+	// Add a temp attribute for sorting logs in simulation output
```

Comment:
The input and output logs would have timestamps, Could we use that instead? Or in the absence of that, we could also use the observed timestamp on record instead of this new input order attribute?